src/main/java/io/jenkins/plugins/checks/github/GitHubChecksPublisher.java

```diff
@@ -39,10 +42,16 @@
      *         a context which contains SCM properties
      */
     public GitHubChecksPublisher(final GitHubChecksContext context, @Nullable final TaskListener listener) {
```

Comment:
Why is listener `Nullable`? There is a `NullListener` if you want no logging.

Reply:
That's because we have to compatible with the old api which doesn't require a listener, we'll remove those null checks for listener after warnings-ng and code-coverage have updated to the new api.

Reply:
code-coverage uses the new API:
https://github.com/jenkinsci/code-coverage-api-plugin/pull/170

Reply:
ok, then I'll open a PR to remove them